clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -5443,6 +5443,13 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
         }
       }
     }
+
+    // At -O0, imply -D__SYCL_DISABLE_PARALLEL_FOR_RANGE_ROUNDING__
```

Comment:
```suggestion
    // At -O0, imply -fsycl-disable-range-rounding
```

We have an option to disable this optimization: `-fsycl-disable-range-rounding`.
As far as I can see all it does is defining the macro, but I guess the option is the intended compiler interface to turn on/off this optimization.

In general, I think this optimization should be off by default for all optimization levels (except maybe O3).

Reply:
Thanks for pointing out the option - I didn't realize that was there.  I will look into adjustments to disable this optimization for all except -O3

Reply:
> I will look into adjustments to disable this optimization for all except -O3

This might be controversial. I suggest making this change in a separate PR. I suppose there are benchmarks where this optimization improves performance and disabling this optimization by default might cause performance degradations (in addition to JIT time improvements). It's better to have a separate commit to handle range rounding defaults.
Tagging @rdeodhar for awareness.

Reply:
I'll create a new PR after this one goes through